src/trigger.js

```diff
@@ -0,0 +1,189 @@
+try {
+  var m = require("mithril");
+} catch (e){
+}
+
+const app = {};
+
+app.view = function(vnode) {
+  const state = vnode.state;
+
+  const matchSearchKey = (project) => {
+    const keys = state.search_key.split(" ");
+    return keys.every((key) => {
+      const r = new RegExp(key, "i");
+      return project.path_with_namespace.match(r);
+    });
+  };
+
+  const projectEvents = (project) => {
+    const project_id = parseInt(project.id);
+    const config_project = state.config_projects[project_id] || {};
+    return config_project.events || {};
+  };
+  const branchList = () => {
+    if (!state.gitlab.branchs) {
+      return m("li", m("div","------"));
+    }
+    return state.gitlab.branchs.filter((branch) =>{ 
+      return branch;
+    }).map((branch) => {
+      return m("li",  m("a", {
+        onclick: (e) => {
+          var targ = e.target;
+          $( "#curBranch" ).html( targ.innerText );
+          e.preventDefault();
+        }
+      }, 
+      branch.name
+      ));
+    });
+  };
+  const branchBtn = () => {
+    return m("div.dropdown",[
+      m('button.btn.btn-primary.btn-block.dropdown-toggle[type="button"][data-toggle="dropdown"]', [
+        "branch ", m("span.caret")
+      ]),
+      m('ul.dropdown-menu[role="menu"]',branchList()),
+    ]);
+  };
+  const tokenList = () => {
+    if (!state.gitlab.triggers) {
+      return m("li.token", m("div","------"));
+    }
+    return state.gitlab.triggers.filter((token) =>{ 
+      return token; 
+    }).map((token) => {
+      return m("li.token", 
+        m("a", {
+          onclick: (e) => {
+            var targ = e.target;
+            $( "#curToken" ).html( targ.innerText );
```

Comment:
Same to  https://github.com/sue445/chrome-gitlab-notifier/pull/124/files#r152980351